addons/phone_validation/models/models.py

```diff
@@ -19,10 +16,60 @@ class BaseModel(models.AbstractModel):
     def _phone_get_number_fields(self):
         """ This method returns the fields to use to find the number to use to
         send an SMS on a record. """
-        return []
+        return [
+            number_fname for number_fname in ('mobile', 'phone') if number_fname in self
+        ]
 
     @api.model
     def _phone_get_country_field(self):
         if 'country_id' in self:
             return 'country_id'
         return False
+
+    def _phone_format(self, fname=False, country=False, force_format='E164', raise_exception=False):
+        """ Format and return number found in a given field e.g. mobile or phone.
+        If fname is not given try to find it based on '_phone_get_number_fields' """
+        if not fname:
+            fnames = self._phone_get_number_fields()
+            fname = fnames[0] if fnames else False
+        if not fname:
+            return False
+        return self._phone_format_number(self[fname], country=country, force_format=force_format, raise_exception=raise_exception)
+
+    def _phone_format_number(self, number, country=False, force_format='E164', raise_exception=False):
+        """ Format and return number according to the asked format.
+
+        :param str number: number to format;
+        :param <res.country> country: country linked to the phone number, used to
+          fetch notably phone_code used in formatting. If not given try to find
+          it based on '_phone_get_country_field';
+        :param str force_format: asked formatting, see phone_validation tools;
+        :param bool raise_exception: raise an exception if cannot format the
+          number: otherwise False is returned;
+        """
+        if not number:
+            return False
+        if len(self) > 1:
```

Comment:
Looks like this method is a bit conflicted with itself as to whether it's a record-based or model-based method.

How about adding a "number=False" parameter inside "_phone_format", and make this one the model-based method, using the phone number if provided else falling back on "_phone_get_number_fields", and also do the country fallback based on record company.

Whereas "_phone_format_number" would be a pure model-based method.
(But in that case maybe it's not useful anymore, as you could directly call "phone_validation.phone_format") ?

Not 100% sure, but it feels a bit wrong currently.